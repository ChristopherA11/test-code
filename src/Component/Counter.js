import React, { useState } from 'react'

const Counter = () => {
    const [count, setCount] = useState(3)
    const incrementCounter = () => {
        setCount((prevCount) => prevCount + 1)
    }
    const decrementCounter = () => {
        setCount((prevCount) => prevCount - 1 )
    }

  return (
    <div>
      <button data-testid = "increment" onClick={incrementCounter}>+</button>
      <p data-testid = "counter">count:{count}</p>
      <button data-testid = "decrement" onClick={decrementCounter}>-</button>
    </div>
  )
}

export default Counter
