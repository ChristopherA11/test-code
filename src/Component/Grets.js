import React from 'react'

const Grets = ({name}) => {
  return (
    <div>
      {name}
    </div>
  )
}

export default Grets;
