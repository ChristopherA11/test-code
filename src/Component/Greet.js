import React from 'react'
import Grets from './Grets'


const Greet = () => {
  return (
    <div>
      Hello
      <Grets 
      title= "John"/>
    </div>
  )
}

export default Greet



