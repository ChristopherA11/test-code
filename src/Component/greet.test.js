import {fireEvent, render, screen} from "@testing-library/react"
import renderer from 'react-test-renderer'
import Greet from "./Greet"
import Grets from "./Grets"
import Counter from "./Counter"


describe("Counter", () => {
   
    test("increment counter", () => {
        render(<Counter />)
    
        const counter = screen.getByTestId("counter")
        const incrementBtn = screen.getByTestId("increment")
    
        fireEvent.click(incrementBtn)
    
        expect(counter).toHaveTextContent("3")
    })
    test("decrement counter", () => {
        render(<Counter />)
    
        const counter = screen.getByTestId("counter")
        const decrementBtn = screen.getByTestId("decrement")
    
        fireEvent.click(decrementBtn)
    
        expect(counter).toHaveTextContent("1")
    })

})
test('Greet renders correctly', () => {
    render(<Greet />)
   const eleId = screen.getByText(/Hello/i)
   expect(eleId).toBeInTheDocument()
})

it("greet check the name", () => {
    render(<Grets name = "john" />)
    const nameStore = screen.getByText("john")
    expect(nameStore).toBeInTheDocument()
})

describe("Counter SnapShot",() => {
    test ("should matches dom Snapshot", () => {
        const tree = renderer.create(<Counter />).toJSON();
        expect(tree).toMatchSnapshot()
    })
})


