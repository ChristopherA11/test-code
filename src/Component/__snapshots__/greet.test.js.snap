// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter SnapShot should matches dom Snapshot 1`] = `
<div>
  <button
    data-testid="increment"
    onClick={[Function]}
  >
    +
  </button>
  <p
    data-testid="counter"
  >
    count:
    2
  </p>
  <button
    data-testid="decrement"
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
